src/cmd/node/rpc/estimate_gas_publish.rs

```diff
@@ -0,0 +1,76 @@
+use anyhow::{Context, Result};
+use clap::Parser;
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::http_client::HttpClientBuilder;
+use jsonrpsee::rpc_params;
+use move_core_types::vm_status::StatusCode;
+use serde::Deserialize;
+use std::fmt;
+use std::path::{Path, PathBuf};
+
+/// Estimate gas for publishing modules.
+#[derive(Parser, Debug)]
+#[clap(about = "Estimate gas for publishing modules")]
+pub struct EstimateGasPublishModule {
+    #[clap(short, long, help = "Account ID in the SS58 format")]
+    account_id: String,
+    #[clap(short, long, help = "Path to the module (compiled by the smove)")]
+    module_path: PathBuf,
+}
+
+/// Gas estimation information.
+#[derive(Deserialize)]
+struct Estimation {
```

Comment:
I'm not sure if there is a need to create new type very similar to `VmResult` type.
Maybe it would be enough to make a type alias to reflect its meaning in this code.

Reply:
As we discussed on Slack
We'll keep it as it is for now... And possibly refactor it in the future.